src/writer/vote.ts

```diff
@@ -48,6 +48,17 @@ export async function verify(body): Promise<any> {
       return Promise.reject('invalid choice');
   }
 
+  // Check if voter already voted
+  const votes = await db.queryAsync(
+    'SELECT created FROM votes WHERE voter = ? AND proposal = ? AND space = ? ORDER BY created DESC LIMIT 1',
+    [body.address, msg.payload.proposal, msg.space]
+  );
+
+  // Reject vote with later timestamp
+  if (votes[0]?.created > msgTs) {
+    return Promise.reject('already voted at later time');
+  }
```

Comment:
It is already checked in action right https://github.com/snapshot-labs/snapshot-sequencer/blob/use-alias-schema/src/writer/vote.ts#L148 ? maybe we should remove that if we already check it in verify

Reply:
I think the check inside `action` is to there in case some other threads/container are processing the same exact request, and is still needed, since we can't guarantee that the check inside `verify` is valid.

Reply:
mm better to handle parallel requests in requestDeduplicator, because this can cause duplicate query and code, meaning additional time to vote for users 🙈

Reply:
We should move all inside `verify`, after implementing requestDeduplicator #202

Reply:
mmm better to install deduplication first